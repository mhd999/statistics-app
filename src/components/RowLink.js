// @flow

import React from 'react';
import styler from 'react-styling';

type Props = {
    link: string,
    text: string
};

const style = styler `
  LinkStyle
    font-size: 12px
    color: #000000
    text-decoration: underline
`;

const RowLink = (props: Props) => (
  <a href={props.link} target="_blank" style={style.LinkStyle}>{props.text}</a>
);

export default RowLink;