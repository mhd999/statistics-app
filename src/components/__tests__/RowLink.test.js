import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import RowLink from '../RowLink';

describe('<RowLink />', () => {
  it('renders a link', () => {
    const component = shallow(
      <RowLink link='http://localhost:80' text='new page'/>
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
