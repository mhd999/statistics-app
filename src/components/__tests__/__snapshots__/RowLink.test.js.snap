exports[`<RowLink /> renders a link 1`] = `
<a
  href="http://localhost:80"
  style={
    Object {
      "color": "#000000",
      "fontSize": "12px",
      "textDecoration": "underline",
    }
  }
  target="_blank">
  new page
</a>
`;
